tests/test.js

```diff
@@ -438,4 +438,173 @@ test('NbsDecoder.getPackets() returns the last packet of each type when given a
+    tsToBigInt(start),
+    'With steps = 0, input start timestamp should be equal to the returned timestamp for single type'
+  );
+});
+
+test('NbsDecoder.nextTimestamp() Stepping forward by one timestamp returns a timestamp which is greater than input.', () => {
+  const [start, end] = decoder.getTimestampRange({ type: pingType, subtype: 0 });
+  const singleTypeTimestamp = decoder.nextTimestamp(start, { type: pingType, subtype: 0 }, 1);
+  const multiTypeTimestamp = decoder.nextTimestamp(start, multiTypeNextTimestampArray, 1);
+
+  assert.ok(
+    tsToBigInt(singleTypeTimestamp) > tsToBigInt(start),
+    'The new return Nbs timestamp is greater than the input timestamp'
+  );
+
+  assert.ok(
+    tsToBigInt(multiTypeTimestamp) > tsToBigInt(start),
+    'The new return Nbs timestamp is greater than the input timestamp'
+  );
+});
+
+test(`NbsDecoder.nextTimestamp() Stepping forward by five returns the expected timestamp for single and multi type`, () => {
+  const [start, end] = decoder.getTimestampRange({ type: pingType, subtype: 0 });
+  const singleTypeTimestamp = decoder.nextTimestamp(start, { type: pingType, subtype: 0 }, 5);
+  const multiTypeTimestamp = decoder.nextTimestamp(start, multiTypeNextTimestampArray, 5);
+
+  assert.equal(tsToBigInt(singleTypeTimestamp), tsToBigInt({ seconds: 1015, nanos: 0 }));
+  assert.equal(tsToBigInt(multiTypeTimestamp), tsToBigInt({ seconds: 1013, nanos: 0 }));
+});
+
+test('NbsDecoder.nextTimestamp() Stepping backward returns a timestamp which is less than input', () => {
+  const [start, end] = decoder.getTimestampRange({ type: pingType, subtype: 0 });
+  const singleTypeTimestamp = decoder.nextTimestamp(end, { type: pingType, subtype: 0 }, -1);
+
+  const multiTypeTimestamp = decoder.nextTimestamp(end, multiTypeNextTimestampArray, -1);
+
+  assert.ok(
+    tsToBigInt(singleTypeTimestamp) < tsToBigInt(end),
+    'The return Nbs timestamp is less than the input timestamp for a single type'
+  );
+
+  assert.equal(tsToBigInt(singleTypeTimestamp), tsToBigInt({ seconds: 1894, nanos: 0 }));
+  ('The return Nbs timestamp exactly matches the expected timestamp');
+
+  assert.ok(
+    tsToBigInt(multiTypeTimestamp) < tsToBigInt(end),
+    'The return Nbs timestamp is less than the input timestamp for multiple types'
+  );
+});
+
+test('NbsDecoder.nextTimestamp() Stepping backwards/forwards respective of begin/end iterators return the correct timestamps.', () => {
+  const [start, end] = decoder.getTimestampRange({ type: pingType, subtype: 0 });
+  const singleTypeTimestampStart = decoder.nextTimestamp(
+    start,
+    { type: pingType, subtype: 0 },
+    -9000
+  );
+  const singleTypeTimestampEnd = decoder.nextTimestamp(end, { type: pingType, subtype: 0 }, 9000);
+
+  assert.equal(tsToBigInt(start), tsToBigInt(singleTypeTimestampStart));
+  assert.equal(tsToBigInt(end), tsToBigInt(singleTypeTimestampEnd));
+});
+
+test('NbsDecoder.nextTimestamp() Multiple cases where timestamp is before beginning', () => {
```

Comment:
A test should test one thing so when it fails it tells you what failed.
Split these tests into individual tests